tools/osbuild-image-info

```diff
@@ -2567,6 +2571,52 @@ def partition_is_lvm(part: Dict) -> bool:
     return part["type"].upper() in ["E6D6D379-F507-44C2-A23C-238F2A3DF928", "8E"]
 
 
+def find_fstab_root(tree, fstype):
+    """
+    Find the root volume under tree by searching for /etc/fstab.
+    This function first checks if the path <tree>/etc/fstab exists and then, if the fstype is btrfs, checks all
+    subvolumes as well.
+
+    Returns None if fstab is not found.
+    """
+    if os.path.exists(f"{tree}/etc/fstab"):
+        return tree
+
+    if fstype == "btrfs":
+        root_subvol = find_root_subvol(tree)
+        if root_subvol:
+            return root_subvol
+
+    return None
+
+
+def find_root_subvol(root):
+    """
+    Given a btrfs volume root, find the subvolume that contains the root OS tree.
+    """
+    def subvol_list_parser(output):
+        """
+        Parse the output of 'btrfs subvolume list' and return just the subvolume names/paths.
+        """
+        paths = []
+        for line in output.strip().split("\n"):
+            # subvolume names can have spaces in them, but they are the last field and they are preceded by the word
+            # path
+            parts = line.split(" ")  # last component is the path
```

Comment:
(super nitpick): this could be written as:
```python
path = line.partition(" path ")[2]
```
which is slightly shorter but maybe too magic (idk, I really like str.partition()).

Reply:
Yeah that works.  I was thinking that this would trip up on the valid (yet very contrived) case where there's a subvolume with the word "path" in it, but it works:
```python
In [1]: line = "ID 34860 gen 2585146 top level 27185 path path to subvol"

In [2]: line.split(" ")
Out[2]:
['ID',
 '34860',
 'gen',
 '2585146',
 'top',
 'level',
 '27185',
 'path',
 'path',
 'to',
 'subvol']

In [3]: line.split(" ").index("path")
Out[3]: 7

In [4]: line.split(" ")[7:]
Out[4]: ['path', 'path', 'to', 'subvol']

In [5]: line.partition(" path ")
Out[5]: ('ID 34860 gen 2585146 top level 27185', ' path ', 'path to subvol')
```

Reply:
Sidenote: `btrfs` has a `--json` flag but it's not supported for `btrfs subvolume list` -_-

Reply:
Changed to use `partition`.